addOns/pscanrulesBeta/src/main/java/org/zaproxy/zap/extension/pscanrulesBeta/InformationDisclosureSuspiciousComments.java

```diff
@@ -125,24 +128,29 @@ private void raiseAlert(HttpMessage msg, int id, String detail) {
         parent.raiseAlert(id, alert);
     }
 
-    private List<Pattern> getPatterns() {
-        if (patterns == null) {
-            patterns = new ArrayList<>();
-            String line = null;
-            File f = new File(Constant.getZapHome() + File.separator + databaseErrorFile);
+    private List<Pattern> setPatterns() {
+        if (PATTERNS == null) {
+            List<Pattern> newPatterns = new ArrayList<>();
+
             BufferedReader reader = null;
             try {
-                reader = new BufferedReader(new FileReader(f));
+                String line = null;
+                URL suspiciousCommentsFile = getClass().getResource(suspiciousCommentsListFile);
+                if (suspiciousCommentsFile == null) {
+                    throw new IOException("Couldn't find resource: " + suspiciousCommentsListFile);
+                }
+                reader = new BufferedReader(new InputStreamReader(suspiciousCommentsFile.openStream()));
                 while ((line = reader.readLine()) != null) {
+                    line = line.trim(); // in case someone puts trailing whitespace in the suspicious cmts file by accident
```

Comment:
The comment can be removed.

Reply:
Why remove explanatory comments? The serious LACK of comments in the existing scanners was less than helpful...

Reply:
In this case it isn't a difficult stacked or combined statement or anything. Use of trim is kind of straight forward and self evident. (At least that's my guess.)

Reply:
(IMHO) The statement speaks for itself: "I don't want leading and trailing whitespace characters in the lines used to create the patterns.". The comment just tells half the story, what about the other implications that the trim has? Lines with just whitespace characters will be ignored and that the whitespace characters are actually redundant because the pattern will be surrounded by word boundary... Why is the accidental case more important than the others? All unnecessary, if you read the code it explains all that.

If you think the whole logic needs to be explained then JavaDoc is better place than appendix comments (which more often than not end up lying and contradict what the code is doing).